api/konnect/v1alpha1/konnect_extension_types.go

```diff
@@ -0,0 +1,176 @@
+// extension by the DataPlane, ControlPlane or GatewayConfiguration APIs.
+// If one of the above mentioned resources successfully refers a KonnectExtension, the underlying
+// deployment(s) spec gets customized to include the konnect-related configuration.
+//
+// +genclient
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:object:generate=true
+// +kubebuilder:subresource:status
+// +kubebuilder:validation:XValidation:rule="oldSelf.spec.controlPlaneRef == self.spec.controlPlaneRef", message="spec.controlPlaneRef is immutable."
+// +kubebuilder:validation:XValidation:rule="self.spec.controlPlaneRef.type == 'konnectID' ? has(self.spec.konnect) : true",message="konnect must be set when ControlPlaneRef is set to KonnectID."
+// +kubebuilder:validation:XValidation:rule="self.spec.controlPlaneRef.type == 'konnectNamespacedRef' ? !has(self.spec.konnect) : true",message="konnect must be unset when ControlPlaneRef is set to konnectNamespacedRef."
+// +apireference:kgo:include
+// +kong:channels=gateway-operator
+type KonnectExtension struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of the KonnectExtension resource.
+	Spec KonnectExtensionSpec `json:"spec,omitempty"`
+	// Status is the status of the KonnectExtension resource.
+	Status KonnectExtensionStatus `json:"status,omitempty"`
+}
+
+// KonnectExtensionList contains a list of KonnectExtension.
+//
+// +kubebuilder:object:root=true
+// +apireference:kgo:include
+type KonnectExtensionList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []KonnectExtension `json:"items"`
+}
+
+// KonnectExtensionSpec defines the desired state of KonnectExtension.
+type KonnectExtensionSpec struct {
+	// ControlPlaneRef is a reference to a Konnect ControlPlane this KonnectExtension is associated with.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self.type != 'kic'", message="kic type not supported as controlPlaneRef."
+	ControlPlaneRef commonv1alpha1.ControlPlaneRef `json:"controlPlaneRef"`
+
+	// DataPlaneClientAuth is the configuration for the client certificate authentication for the DataPlane.
+	// In case the ControlPlaneRef is of type KonnectID, it is required to set up the connection with the
+	// Konnect Platform.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default={certificateSecret:{provisioning: Automatic}}
+	// +kubebuilder:validation:XValidation:rule="self.certificateSecret.provisioning == 'Manual' ? has(self.certificateSecret.secretRef) : true",message="secretRef must be set when provisioning is set to Manual."
+	// +kubebuilder:validation:XValidation:rule="self.certificateSecret.provisioning == 'Automatic' ? !has(self.certificateSecret.secretRef) : true",message="secretRef must not be set when provisioning is set to Automatic."
```

Comment:
Sorry for not being clear on this: I meant to put these on the `DataPlaneClientAuth` type. Does that makes sense to you?

Reply:
sure, moved